src/libponyc/expr/operator.c

```diff
@@ -480,6 +480,18 @@ bool expr_assign(pass_opt_t* opt, ast_t* ast)
       }
     }
 
+    if(ast_child(left) != NULL)
+    {
+      ast_error(opt->check.errors, ast,
+        "not safe to write right side to left side");
+      ast_error_continue(opt->check.errors, r_type, "right side type: %s",
+        ast_print_type(r_type));
+      ast_error_continue(opt->check.errors, ast_child(left), "left side type: %s",
+        ast_print_type(ast_type(ast_child(left))));
+      ast_free_unattached(wl_type);
+      return false;
+    }
+
     ast_error(opt->check.errors, ast,
       "not safe to write right side to left side");
     ast_error_continue(opt->check.errors, wl_type, "right side type: %s",
```

Comment:
Could this be wrong? I guess wl_type is the left type, but the error says it's "right side type".

Reply:
Can you explain it? :)

Reply:
It's the left side of the expression that has the result you are going to assign in the expression in question. It's a bit of indirection.